circuits.go

```diff
@@ -196,6 +198,29 @@ type PubSignals interface {
 	PubSignalsMapper
 }
 
+// StateInfoPubSignals interface implemented by types that can return states info
+type StateInfoPubSignals interface {
+	GetStatesInfo() StatesInfo
+}
+
+// StatesInfo struct. A collection of states and gists
+type StatesInfo struct {
+	States []States
+	Gists  []Gists
+}
+
+// States information
+type States struct {
```

Comment:
@x1m3 @vmidyllic 
Why it is plural `States` and `Gists`?

Reply:
agree, it should be State/ Gist or just anonymus

Reply:
agree. Changed